Sources/ApplicativeRouter/SyntaxRouter.swift

```diff
@@ -159,12 +159,21 @@ private func request(from data: RequestData) -> URLRequest? {
 }
 
 private func request(from data: RequestData, base: URL?) -> URLRequest? {
-  return urlComponents(from: data)
-    .url(relativeTo: base)
-    .map {
-      URLRequest(url: $0)
-        |> \.httpMethod .~ data.method?.rawValue
-        |> \.httpBody .~ data.body
+
+
+
+  // Due to this bug https://bugs.swift.org/browse/SR-6527, if `URLComponents` doesn't contain any path or
+  // query information, it will failure to create a `URL`.
+  return
+    (
+      data.path.isEmpty && data.query.isEmpty
+        ? (base ?? URL(string: "/"))
+        : urlComponents(from: data).url(relativeTo: base)
```

Comment:
this was coming back `nil` on linux, so we have to first check if path+query are empty, and if so just default to the base url.